src/executor/evaluate/mod.rs

```diff
@@ -281,5 +281,10 @@ async fn evaluate_function<'a, T: 'static + Debug>(
 
             Ok(Evaluated::from(Value::Str(converted)))
         }
+        Function::Trim(expr) => match eval_to_str("TRIM", expr).await? {
+            Nullable::Value(string) => Ok(Value::Str(string.trim().to_owned())),
+            Nullable::Null => Ok(Value::Null),
```

Comment:
It seems that missing test case is only this one.
```rust
Nullable::Null => {
    panic!("lol");
}
```

It still pass all tests.
Except for this, it's.. perfect. 👍 

You may feel test cases are not enough but it's really enough.
Many string functions uses `eval_to_str` and it is based on `evaluate`, so it is not necessary to cover all ast expr branches in here.

This can be an one good reason of why closure or function is better than using macro.
Macro shows similar behavior with function in the view of.. reducing redundant codes, but it is only for the human, not the system.

To verify all codes, we need to reach all code branches generated by macros, but we don't need to do that if we use functions.
However, as we all know, there are also many advantages of using macros, so not easy to say what's better I think..

Reply:
Oh no... I'll add a test case for Null.